### ISSUE TITLE
ActionMode is closed when pressing back button while soft keyboard is visible

### ISSUE THREAD

**Issue by rfgamaral** [JakeWharton/ActionBarSherlock#650]
When the soft keyboard is visible on screen, pressing the back button should hide the keyboard and nothing else. However, on my Android 2.1 device, pressing the back button hides the keyboard and closes the active ActionMode.

The same does not happen on my Android 4.0.4 device. When I press the back button, it hides the keyboard and that's it. I press the back button again and now it closes the ActionMode.

I'm assuming this is a bug in ActionBarSherlock because it is happening on 2.1 and not 4.0.4 which is using the native ActionBar. Is this a known issue and is there a workaround?

**Comment by rfgamaral** [comment]
I debugged this a little bit further and here's what I've found:

The `dispatchKeyEvent` is the method responsible for handling key presses, like the back button. When running Android 4.0.4, this method _doesn't_ get called when I press the back button to hide the keyboard, but does get called in the next back button press to go back. However, when running Android 2.1 (and maybe other versions), this method _does_ get called when I press the back button to hide the keyboard and to go back.
